### PR TITLE
K8SPXC-456: trim "/versions" in versionservice url

### PR DESCRIPTION
**Pull request by qJkee**
[K8SPXC-456](https://jira.percona.com/browse/K8SPXC-456)

### CODE REVIEW

**Review by tplavcic — COMMENTED** [review]
The only complaint that I can find to this is that we maybe want to remove the ending `/` also, so if somebody has `https://check.percona.com/versions/` that it removes `/versions/`.
I know we didn't have this ending `/` in PXC cr.yaml example, but we have it in PSMDB: https://github.com/percona/percona-server-mongodb-operator/blob/master/deploy/cr.yaml#L16
Otherwise looks ok.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| haproxy | passed |
| big-data | passed |
| storage | failed |
| scaling | passed |
| scaling-proxysql | failed |
| init-deploy | passed |
| restore-to-encrypted-cluster | passed |
| limits | passed |
| monitoring | failed |
| demand-backup | passed |
| self-healing | passed |
| scheduled-backup | passed |
| upgrade | passed |
| self-healing-advanced | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | failed |
| security-context | passed |
| demand-backup-encrypted-with-tls | passed |
| operator-self-healing | passed |
| affinity | failed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |

**Review by delgod — APPROVED** [review]

File: pkg/controller/pxc/vs.go [hunk 1]
```diff
@@ -21,7 +22,7 @@ func (vs VersionServiceClient) GetExactVersion(endpoint string, vm versionMeta)
 
 	srvCl := versionserviceclient.NewHTTPClientWithConfig(nil, &versionserviceclient.TransportConfig{
 		Host:     requestURL.Host,
-		BasePath: requestURL.Path,
+		BasePath: strings.TrimSuffix(requestURL.Path,"/versions"),
```
Comment (delgod) [inline pkg/controller/pxc/vs.go:25]:
```suggestion
		BasePath: strings.TrimSuffix(strings.TrimSuffix(requestURL.Path,"/"),"/versions"),
```

**Review by delgod — CHANGES_REQUESTED** [review]

**Comment by qJkee** [comment]
> The only complaint that I can find to this is that we maybe want to remove the ending `/` also, so if somebody has `https://check.percona.com/versions/` that it removes `/versions/`.
> I know we didn't have this ending `/` in PXC cr.yaml example, but we have it in PSMDB: https://github.com/percona/percona-server-mongodb-operator/blob/master/deploy/cr.yaml#L16
> Otherwise looks ok.

Yea, it's exists in psmdb, but because we don't change it when it was updated
But while we have one and only one VS URL, we should not do any extra manipulations, which can cause bad effect in future with custom user's URL

File: pkg/controller/pxc/vs.go [hunk 2]
```diff
@@ -21,7 +22,7 @@ func (vs VersionServiceClient) GetExactVersion(endpoint string, vm versionMeta)
```
Comment (delgod) [inline pkg/controller/pxc/vs.go:22]:
```suggestion
if requestURL.Path == "https://check.percona.com/versions" {
    requestURL.Path = "https://check.percona.com"
}
```

**Review by delgod — CHANGES_REQUESTED** [review]

File: deploy/cr.yaml [hunk 3]
```diff
@@ -27,7 +27,7 @@ spec:
 #  pause: false
   updateStrategy: SmartUpdate
   upgradeOptions:
-    versionServiceEndpoint: https://check.percona.com
+    versionServiceEndpoint: https://check.percona.com/versions
```
Comment (delgod) [inline deploy/cr.yaml:30]: lets have correct value in the CR
```suggestion
    versionServiceEndpoint: https://check.percona.com
```

**Review by delgod — CHANGES_REQUESTED** [review]

**Review by delgod — APPROVED** [review]

**Review by tplavcic — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| big-data | passed |
| recreate | passed |
| haproxy | passed |
| storage | failed |
| scaling-proxysql | failed |
| init-deploy | passed |
| restore-to-encrypted-cluster | passed |
| limits | passed |
| monitoring | failed |
| demand-backup | passed |
| self-healing | passed |
| upgrade | passed |
| scheduled-backup | passed |
| upgrade-consistency | passed |
| security-context | passed |
| monitoring-2-0 | failed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| demand-backup-encrypted-with-tls | failed |
| affinity | failed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| haproxy | passed |
| recreate | passed |
| big-data | passed |
| storage | failed |
| scaling-proxysql | failed |
| init-deploy | passed |
| restore-to-encrypted-cluster | passed |
| limits | passed |
| monitoring | failed |
| demand-backup | passed |
| self-healing | passed |
| upgrade | passed |
| self-healing-advanced | passed |
| upgrade-consistency | passed |
| scheduled-backup | passed |
| security-context | passed |
| monitoring-2-0 | failed |
| operator-self-healing | passed |
| demand-backup-encrypted-with-tls | failed |
| affinity | failed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |

**Pull request merged**